transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,219 @@
+const vm = require('vm'),
+      _ = require('underscore'),
+      config = require('../config'),
+      lineage = require('../lib/lineage'),
+      logger = require('../lib/logger'),
+      messages = require('../lib/messages'),
+      utils = require('../lib/utils');
+
+const hasRun = (doc) => Boolean(
+  doc &&
+  doc.transitions &&
+  doc.transitions.multi_form_alerts
+);
+
+const getAlerts = () => config.get('multi_form_alerts');
+
+/* Returned list does not include the change.doc. */
+const fetchReports = (latestTimestamp, timeWindowInDays, formTypes) => {
+  return utils.getReportsWithinTimeWindow(latestTimestamp, timeWindowInDays)
+    .then((reports) => {
+      if (formTypes && formTypes.length) {
+        return reports.filter((report) => report.form && formTypes.includes(report.form));
+      }
+      return reports;
+    })
+    .then(hydrateDocs);
+};
+
+const hydrateDocs = (docs) => {
+  const fetchedDocsPromise = docs.reduce(function(promise, doc) {
+    return promise.then((fetchedDocs) => {
+      return lineage.fetchHydratedDocPromise(doc._id)
+        .then(fetchedDoc => {
+          fetchedDocs.push(fetchedDoc);
+          return fetchedDocs;
+        });
+    });
+  }, Promise.resolve([]));
+  return fetchedDocsPromise;
```

Comment:
I don't think this will perform well if someone has lots of reports. For `n` reports you're making `n` requests to the database. Instead you should write a util to collect up all the parent IDs, unique them, request them all from the DB in one go, and then bind them back into the reports.